examples/raw_stream_into_file.rs

```diff
@@ -0,0 +1,38 @@
+use clickhouse::format::OutputFormat;
+use clickhouse::Client;
+use tokio::fs::File;
+use tokio::io::AsyncWriteExt;
+
+/// An example of streaming the result of a query in an arbitrary format to a file.
+
+#[tokio::main]
+async fn main() -> clickhouse::error::Result<()> {
+    let filename = "output.csv";
+    let client = Client::default().with_url("http://localhost:8123");
+
+    let mut raw_cursor = client
+        .query(
+            "
+                SELECT number, randomPrintableASCII(20)
+                FROM system.numbers
+                LIMIT 100000
+            ",
+        )
+        .fetch_raw(OutputFormat::CSV)?;
+
+    let mut file = File::create(filename).await.unwrap();
+
+    loop {
```

Comment:
See https://github.com/ClickHouse/clickhouse-rs/pull/182#discussion_r1871132609 for how to make this example more canonical.